dcc-portal-ui/app/scripts/downloader/js/request.js

```diff
@@ -136,8 +136,10 @@
 
         $scope.params.processing = false;
       },function (error) {
-        $scope.params.processing = false;
-        $scope.params.downloadEnabled = false;
+        if(error.status === 503){
+          $scope.params.processing = false;
+          $scope.params.downloadEnabled = false;
+        }
```

Comment:
Also, we are not explicitly returning `true` or `false` from:

https://github.com/icgc-dcc/dcc-portal/blob/41fe22825c3c1edef43d6dc53315ac558f606bc1/dcc-portal-ui/app/scripts/app/js/app.js#L602-L614

Does that mean `false`? If it does, it stops the promise chain and `finally`  handlers etc. won't be called.